openarc/openarc-ar.c

```diff
@@ -501,6 +501,9 @@ ares_parse(u_char *hdr, struct authres *ar)
 				if (n > 0)
 					n--;
 
+				prevstate = state;
+				state = 2;
```

Comment:
> State 2 is not terminal, which I believe means "none; dkim=pass ..." would be accepted

That was actually the intention...

We could either 
- keep `state = 2` and in `case 2` add an additional test for `prevstate == 3` or
- set `state = 14` (a new state) which does the following:
```
case 14:
	return -1;	
	break;
```

> You should also add it to the list of fatal states at the end of this function for this to be thorough.

If I add the new state (e.g. 14) to list of fatal states it will always trigger if we encounter a 'none', because after we found 'none', state will be 14 - or in other words: a final state must never be a fatal state...